File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/customcontrols/gamepad/GamepadButton.java
package net.kdt.pojavlaunch.customcontrols.gamepad;

/**
 *  このクラスは、ゲームパッドに物理的に存在するボタンに対応します
 */
public class GamepadButton extends GamepadEmulatedButton {
    public boolean isToggleable = false;
    private boolean mIsToggled = false;
/**
 * 「on Down State Changed」処理を実行します。
 * このメソッドは特定の機能を提供するために実装されています。
 */

    @Override
    protected void onDownStateChanged(boolean isDown) {
        if(isToggleable) {
            if(!isDown) return;
            mIsToggled = !mIsToggled;
            Gamepad.sendInput(keycodes, mIsToggled);
            return;
        }
        super.onDownStateChanged(isDown);
    }
/**
 * 「reset Button State」処理を実行します。
 * このメソッドは特定の機能を提供するために実装されています。
 */
    @Override
    public void resetButtonState() {
        if(!mIsDown && mIsToggled) {
            Gamepad.sendInput(keycodes, false);
            mIsToggled = false;
        } else {
            super.resetButtonState();
        }
    }
}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/ui/subassembly/customcontrols/gamepad/GamepadButton.kt
package com.arata.yukarilauncher.ui.subassembly.customcontrols.gamepad

/**
 * ゲームパッドのトグル可能なボタンを表すクラス。
 * GamepadEmulatedButtonを継承し、トグル動作（押すごとにON/OFF切り替え）をサポートします。
 */
open class GamepadButton : GamepadEmulatedButton() {
    /** トグル動作が有効かどうか。 */
    var isToggleable = false
    private var mIsToggled = false

    override fun onDownStateChanged(isDown: Boolean) {
        if (isToggleable) {
            if (!isDown) return
            mIsToggled = !mIsToggled
            Gamepad.sendInput(keycodes, mIsToggled)
            return
        }
        super.onDownStateChanged(isDown)
    }

    override fun resetButtonState() {
        if (!mIsDown && mIsToggled) {
            Gamepad.sendInput(keycodes, false)
            mIsToggled = false
        } else {
            super.resetButtonState()
        }
    }
}
